pkg/acsengine/defaults.go

```diff
@@ -584,8 +591,22 @@ func setMasterProfileDefaults(a *api.Properties, isUpgrade bool) {
 	}
 }
 
-// setVMSSDefaults
-func setVMSSDefaults(a *api.Properties) {
+// setVMSSDefaultsForMasters
+func setVMSSDefaultsForMasters(a *api.Properties) {
+	if a.MasterProfile.Count > 100 {
+		a.MasterProfile.SinglePlacementGroup = helpers.PointerToBool(false)
+	}
+	if a.MasterProfile.SinglePlacementGroup == nil {
+		a.MasterProfile.SinglePlacementGroup = helpers.PointerToBool(api.DefaultSinglePlacementGroup)
+	}
+	if a.MasterProfile.HasAvailabilityZones() && (a.OrchestratorProfile.KubernetesConfig == nil || a.OrchestratorProfile.KubernetesConfig.LoadBalancerSku == "") {
+		a.OrchestratorProfile.KubernetesConfig.LoadBalancerSku = "Standard"
```

Comment:
if `a.OrchestratorProfile.KubernetesConfig == nil` is true it would cause a nil point dereference.

At this point this should never be true (we should be setting `KubernetesConfig` before we reach this line) but the logic of the if statement is contradictory.